docs/protocol/core/v3.0.rst

```diff
@@ -1129,50 +1129,20 @@ that an implementation of this specification could be modular and
 allow for different store implementations to be used.
 
 The store interface defines a set of operations involving `keys` and
-`values`. In the context of this interface, a `key` is any
-string containing only characters in the ranges ``a-z``, ``A-Z``,
-``0-9``, or in the set ``/.-_``, and a `value` is any sequence of
-bytes. It is assumed that the store holds (`key`, `value`) pairs, with
-only one such pair for any given `key`. I.e., a store is a mapping
-from keys to values. It is also assumed that keys are case sensitive, 
-i.e., the keys "foo" and "FOO" are different.
+`values`. In the context of this interface, a `key` is any string
+containing only characters in the ranges ``a-z``, ``A-Z``, ``0-9``, or
+in the set ``/.-_``, where the final character is **not** a ``/``
+character. A `value` is any sequence of bytes.
 
-A store can make the following assumption on the structures of the keys it will receive:
-
-- A key always:
-  - start with ``meta/``
-  - start  with ``data/``
-  - is exactly ``zarr.json``.
-
-- Most of the keys:
-  - start with ``meta/root``
-  - start with ``data/root``
-
-
-- List operations ``list_dir`` will always be passed keys ending with a trailing
-  slash, that is to say it will only be asked to work with complete node names.
-
-Store implementation can assume they will only be given trailing slashes, and
-protocol implementation MUST pass trailing slashes to underlying stores.
-
-For example, a store containing the following keys:
-
- - ``meta/root/2018/.group``
- - ``meta/root/2018-01/.group``
- - ``meta/root/2018/bar/.array``
- - ``data/root/2018/bar/0.0``
-
-The following queries are invalid:
-  - ``list_dir('201')`` is invalid as ``"201"`` is not an existing node.
-  - ``list_dir('2018')`` is invalid queries as ``"2018"`` does not ends with a ``/``,
-
-This is valid:
-  - ``list_dir('2018/')``
-  - ``list_dir('2018-01/')``
-
-This allows store implementation to avoid having to check for trailing slashes,
-and avoid issues like "list_dir('2018')" returning values likes ``-01``
```

Comment:
I removed this because I thought it would be simpler to  just state that, for this protocol, by definition a prefix is always a string with a trailing slash, see below.